apps/server/src/modules/user-login-migration/service/user-migration.service.ts

```diff
@@ -15,6 +15,8 @@ export class UserMigrationService {
 	) {}
 
 	async migrateUser(currentUserId: EntityId, externalUserId: string, targetSystemId: EntityId): Promise<void> {
+		await this.userService.findByExternalId(externalUserId, targetSystemId);
```

Comment:
Little bit weird to look for the user and doing nothing  with it...
Maybe you could move this call in a seperate private function and call it something like checkForExternalIdDuplicatesAndThrow or something like that